### ISSUE TITLE
Confirm compatibility with WordPress 6.6

### ISSUE THREAD

**Issue by joelabreo227** [rtCamp/nginx-helper#346]
## Summary

Confirm compatibility with WordPress 6.6.

### Team
Primary Assignee: @pratik-londhe4 
PR reviewer: @Pathan-Amaankhan 

## References

- https://make.wordpress.org/core/6-6/

## Acceptance Criteria

- As a user, I can use Nginx Helper with WordPress 6.6 without any compatibility issues
- New features work as expected
- Deprecations are gracefully handled

**Comment by pratik-londhe4** [comment]
# Nginx Helper Plugin Test Report 🚧 

###  Local Config
OS: Ubuntu 24.04LTS 
Site: WP v6.6 setup using EE4
Cache: Redis cache


> [!WARNING] 
> Below test cases are only tested with Redis Cache method
>

| Test Scenario                              | Description                                                                 | Expected Outcome                                      |
|--------------------------------------------|-----------------------------------------------------------------------------|-------------------------------------------------------|
| **Purging Conditions**                     |                                                                             |                                                       |
| Purge Homepage on Post Edit                | Check if homepage cache purges when a post/page/custom post is modified or added. | Cache for the homepage is purged. ✅                     |
| Purge Homepage on Post Delete              | Check if homepage cache purges when a published post/page/custom post is trashed. | Cache for the homepage is purged.   ✅                  |
| Purge Post/Page on Publish                 | Verify cache purging when a post is published.                              | Cache for the published post/page is purged.  ✅         |
| Purge on New Comment                       | Check cache purging when a comment is approved/published.                   | Cache for the post/page with the new comment is purged. ✅  |
| Purge on Deleted Comment                   | Verify cache purging when a comment is unapproved/deleted.                  | Cache for the post/page with the deleted comment is purged. ✅  |
| Purge Archives on Post Edit                | Check if archives cache purges when a post/page/custom post is modified or added. | Cache for  archives is purged. ✅ 
| Purge Archives on Post Delete              | Verify archives cache purging when a published post/page/custom post is trashed. | Cache for archives is purged.  ✅                         |
| Purge Archives on New Comment              | Check cache purging when a comment is approved/published in archives.       | Cache for archives with new comment is purged.  ✅       |
| Purge Archives on Deleted Comment          | Verify cache purging when a comment is unapproved/deleted in archives.      | Cache for archives with deleted comment is purged. ✅    |
| Purge Feeds                                | Ensure feed caches are purged along with posts and pages.                   | Cache for feeds is purged regardless if it is enabled or not.   ⚠️                        |
| Custom Purge URL                           | Verify custom purge URLs functionality.                                     | Custom URLs are purged as specified.    ✅               |
---------

- in settings it asks to not put domain in the Custom Purge URL option but no warning/ error is shown if we add URLs with domain ( `eg. https://google.com` )
- On cliking purge all cache, it just says purge initiated, no confirmation if it was ever completed.
- No error/warning for entering wrong redis config

@Pathan-Amaankhan 🔽 
- LMK if we have access to nginx with FastCGI cache setup on a server, so I can test for it

cc: @joelabreo227 @Lokesh-Mangyani17

**Comment by Pathan-Amaankhan** [comment]
@pratik-londhe4 
I don't think we have a site with FastCGI cache setup for testing purposes, but you can try creating one on https://easydash.rtcamp.com/. 👍 

Ref video: https://www.youtube.com/watch?v=QhYMdpHwyXA
Ref doc: https://easyengine.io/wordpress-nginx/tutorials/single-site/fastcgi-cache-with-purging/

cc: @Lokesh-Mangyani17
